### ISSUE TITLE
Make most shortcuts customizable

### ISSUE THREAD

**Issue by ouuan** [cpeditor/cpeditor#214]
**Is your feature request related to a problem? Please describe.**

Now only a few shortcuts are customizable.

**Describe the solution you'd like**

After #166 is done, it will be possible to make most shortcuts customizable and override the default shortcuts.

**Describe alternatives you've considered**

N/A

**Additional context**

The shortcuts that won't be customizable include some Qt shortcuts like <kbd>Ctlr+Tab</kbd>/<kbd>Ctrl+Shift+Tab</kbd> to switch between tabs, `QTextEdit` shortcuts, and <kbd>Tab</kbd>(with a selection)/<kbd>Shift+Tab</kbd> for indentation.

**Comment by cs-mshah** [comment]
additional shortcuts like ctrl+(+/-) for increasing decreasing font size and many other shortcuts from other editors like sublime text. This editor can have shortcuts for everything like 'submit' and can be made so that minimum or no use of mouse is required.

**Comment by cs-mshah** [comment]
@ouuan Could you add shortcuts like crtrl+D to select next occurrence of the selected word. This is **REALLY NEEDED** to change something like datatypes form ll to int or vice versa.  
Currently ctrl+D is having some funny effect and it deletes the selected word.

**Comment by ouuan** [comment]
> @ouuan Could you add shortcuts like crtrl+D to select next occurrence of the selected word. This is **REALLY NEEDED** to change something like datatypes form ll to int or vice versa.
> Currently ctrl+D is having some funny effect and it deletes the selected word.

I don't see it's related to this thread. This thread is for making shortcuts for existing actions customizable, not for creating new actions and binding shortcuts for them.

For the use case mentioned in your request, why don't you use Find & Replace?

**Comment by fengwei2002** [comment]
Is it possible to make a switch that allows the user to use three different shortcuts to switch between three different views?

For example, when half the screen is used, you can use 1 2 instead of 3
When using full screen use 1 3 instead of 2

What I mean is that we can use three different shortcut keys to quickly switch from one layout to another layout, without having to combine the switching shortcut keys of the three layouts

**Comment by huaruoji** [comment]
Yes, I would very much like to be able to delete a line with ctrl+d
